src/lib/homeobject_impl.hpp

```diff
@@ -94,7 +91,7 @@ class HomeObjectImpl : public HomeObject,
     std::map< pg_id_t, shared< PG > > _pg_map;
 
     mutable std::shared_mutex _shard_lock;
-    std::map< shard_id_t, ShardIterator > _shard_map;
+    std::map< shard_id_t, ShardPtr > _shard_map;
```

Comment:
There doesn't seem to be any reason for this change that I can see? This is more expensive; so we should maintain current code if possibe.

Reply:
ok, will change back.